### PR TITLE
Allow selecting multiple exporters

### LINKED ISSUE #1755: Selecting multiple exporters via environmental variables
## What are you trying to achieve?

The [exporter selection environmental variables](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#exporter-selection) allow only to select a SINGLE one. It would be handy to allow selecting more than one e.g. `otlp,console`.

## Additional context

An issue where multiple exporters were also mentioned: 
- https://github.com/open-telemetry/opentelemetry-specification/issues/1309
- https://github.com/open-telemetry/opentelemetry-specification/pull/1318

### Allow adding more than one exporter

We can allow adding multiple exporters separated by `,`  via `OTEL_TRACES_EXPORTER` and `OTEL_METRICS_EXPORTER`.

The only problem is that it is not a forward-compatible change. However, it is backward-compatible.

Example: `OTEL_TRACES_EXPORTER=zipkin,otlp`

### Add new env vars 

We can add new  `OTEL_TRACES_EXPORTERS` and `OTEL_METRICS_EXPORTERS` env vars which could contain a comma-separated list of exporters.

This change is forward-compatible, however, it makes the API/specs longer. Moreover e,.g. if both `OTEL_TRACES_EXPORTER` and `OTEL_TRACES_EXPORTERS` are specified then the "result" should be concatenated. 

Example: if `OTEL_TRACES_EXPORTER=jaeger` and  `OTEL_TRACES_EXPORTERS=zipkin,otlp` then we should add all 3 exporters.

### PR DESCRIPTION
**Pull request by pellared**
Fixes #1755

## Changes

- Allow selecting multiple exporters via `OTEL_TRACES_EXPORTER` and `OTEL_METRICS_EXPORTER`  by using a comma-separated list.

The [exporter selection environmental variables](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#exporter-selection) allow only to select a SINGLE one. It would be handy to allow selecting more than one e.g. `otlp,console`.

### CODE REVIEW

**Review by yurishkuro — APPROVED** [review]

**Comment by carlosalberto** [comment]
Maybe totally straightforward, but I'd like to see a prototype of this in any language.

**Comment by SergeyKanzhelev** [comment]
I think the scenario in PR description `otlp,console` totally makes a good use cases. In production cases there will be situations when different exporters will need different processors - like a sampling or different aggregations. I suggest we  stop at multiple exporters for now in environment variables.

@carlosalberto has a good suggestion. I think it will be trivial to implement in .NET, can you please demonstrate the implementation - issue https://github.com/open-telemetry/opentelemetry-dotnet/issues/1453?

**Comment by pellared** [comment]
@carlosalberto @SergeyKanzhelev 

> @carlosalberto has a good suggestion. I think it will be trivial to implement in .NET, can you please demonstrate the implementation - issue [open-telemetry/opentelemetry-dotnet#1453](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1453)?

I do not think it is trivial. Especially that .NET has almost no support for env vars at the moment. References:
- https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md#environment-variables
- https://github.com/open-telemetry/opentelemetry-dotnet/issues/1453#issuecomment-858454790

I feel that the design may be similar to `OTEL_PROPAGATORS`. I made a quick look at the [Python SDK](https://github.com/open-telemetry/opentelemetry-python/). The `OTEL_PROPAGATORS` implementation looks nice and probably a similar pattern could be used for exporters. WDYT @owais?

I have no idea what is the ETA for implementing https://github.com/open-telemetry/opentelemetry-dotnet/issues/1453. Of course. However, I see no problem having this PR open until someone actually has time to implement it (in whatever language) 😉

**Comment by jkwatson** [comment]
For what it's worth, we implemented this for trace exporters in Java before the env var spec got finalized, at which point we pulled it back to a single exporter. It's a little tricky, however, since just specifying an exporter doesn't determine what span processor you might want to use. And, for a `logging` exporter, a batch span processor probably doesn't make much sense. 

In the otel java instrumentation agent, we automatically add the logging exporter if the user enables `debug` mode for the javaagent. And, in that case, we wire it up with a SimpleSpanProcessor, as that makes more sense for the logging use-case.

**Comment by owais** [comment]
> Maybe totally straightforward, but I'd like to see a prototype of this in any language.

Python already supports this exactly as proposed and it works quite well. 

https://github.com/open-telemetry/opentelemetry-python/blob/b3455cd1164f9c5f336cc26a52fb351cb422b0b2/opentelemetry-distro/src/opentelemetry/distro/__init__.py#L78-L82
https://github.com/open-telemetry/opentelemetry-python/blob/b3455cd1164f9c5f336cc26a52fb351cb422b0b2/opentelemetry-distro/src/opentelemetry/distro/__init__.py#L141-L146
https://github.com/open-telemetry/opentelemetry-python/blob/b3455cd1164f9c5f336cc26a52fb351cb422b0b2/opentelemetry-distro/src/opentelemetry/distro/__init__.py#L48

File: specification/sdk-environment-variables.md [hunk 1]
```diff
@@ -132,13 +132,15 @@ thrift or protobuf.  As of 1.0 of the specification, there
 
 **Status**: [Stable](document-status.md)
 
-We define environment variables for setting a single exporter per signal.
+We define environment variables for setting an exporter per signal.
```
Comment (tigrannajaryan) [inline specification/sdk-environment-variables.md:135]:
```suggestion
We define environment variables for setting one or more exporters per signal.
```

**Review by owais — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
@owais what's the behavior of exporters w.r.t. @jkwatson's [comment](https://github.com/open-telemetry/opentelemetry-specification/pull/1758#issuecomment-860324182)? Will console exporter will be added with the batch processor? Is there anything worth noting in this PR?

**Comment by owais** [comment]
@SergeyKanzhelev @jkwatson  Python implement does not treat console or any other exporter as a special case. It always uses batch exporter. Ability to quickly add console exporter without removing a real exporter is already immensely useful even if the output is a bit delayed because of batching. I have not seen this cause any issues in practice. In fact it actually is quite nice since you can predict exactly when a remote exporter should export the spans. 

Besides, console exporter is not the only use case as this can be used to export to multiple remote exporters as well.

IMO trying to specify multiple combinations of process+exporter via env vars will get too complex and we should expect users to setup such pipelines with code instead.

**Comment by carlosalberto** [comment]
From the Spec call yesterday: although supporting multiple tracer exporters is recurrent request, supporting multiple metrics exporters is not a clear thing (e.g. pull vs push based systems). 

Moreover, this is closely related to the effort by the new Instrumentation group/SIG so let's hold this a little bit for now, while we wait for such group to review it.

**Comment by pellared** [comment]
> From the Spec call yesterday: although supporting multiple tracer exporters is recurrent request, supporting multiple metrics exporters is not a clear thing (e.g. pull vs push based systems).

Is there any problem to make them working "side-by-side"? I think that each concrete exporter should have its own set of environmental variables to configure it.

> Moreover, this is closely related to the effort by the new Instrumentation group/SIG so let's hold this a little bit for now, while we wait for such group to review it.

👍

**Comment by jkwatson** [comment]
> @carlosalberto
> 
> > From the Spec call yesterday: although supporting multiple tracer exporters is recurrent request, supporting multiple metrics exporters is not a clear thing (e.g. pull vs push based systems).
> 
> Is there any problem to make them working "side-by-side"? I think that each concrete exporter should have its own set of environmental variables to configure it.

It might not be a problem eventually, but it's certainly not something that is spec'd yet for the metrics SDK, so we shouldn't be providing any configuration for it, IMO.

**Review by reyang — APPROVED** [review]

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
@carlosalberto suggested to wait for instrumentation group to pick this PR up. Thinking about it, this is a small adjustment that doesn't solve problems like different processors and things like this. Just a convenience improvement. So I'd vote to just accept this proposal. Unlikely it will block any sophisticated solution on telemetry pipelines configuration that instrumentation group will come up with. Thoughts?

**Review by jmacd — APPROVED** [review]

**Comment by jmacd** [comment]
@SergeyKanzhelev and @carlosalberto I'm not sure how this belongs with an instrumentation group. Isn't this about SDK configuration? I would assume that many defaults will apply if I set `OTLP_METRICS_EXPORTER=otlp,prometheus` and that essential configuration items might get dedicated environment variables (e.g., `OTLP_METRICS_PROMETHEUS_PORT`, etc.)

**Comment by SergeyKanzhelev** [comment]
I don't have any more concerns and @carlosalberto didn't repond for a while, I assume it's ok. Merging.

**Pull request merged**